api/categorisation.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabularmetadata"
+	"github.com/pkg/errors"
+)
+
+/*
+CheckDefaultCategorisation checks the default categorisation of a given dimension
+so that we store the correct parent dimension for a given set of options?
+*/
+func (api *API) CheckDefaultCategorisation(dimName string, datasetName string) (string, string, error) {
+
+	ctx := context.Background()
+	labelMap := make(map[string]string)
+	cats, err := api.ctblr.GetCategorisations(ctx, cantabular.GetCategorisationsRequest{
+		Dataset:  datasetName,
+		Variable: dimName,
+	})
+	if err != nil {
+		return "", "", err
+	}
+
+	names := make([]string, 0)
+
+	if len(cats.Dataset.Variables.Edges) > 0 {
```

Comment:
No need to check len(slice/map) before ranging. If len == 0 then block will be skipped